docs/index.asciidoc

```diff
@@ -129,6 +133,50 @@ The number of seconds to wait after failure before retrying.
 A MongoDB URI to connect to.
 See http://docs.mongodb.org/manual/reference/connection-string/.
 
+[id="plugins-{type}s-{plugin}-action"]
+===== `action`
+
+* Value type is <<string,string>>
+* Default value is `insert`.
+
+The method used to write processed events to MongoDB.
+
+Possible values are `insert`, `update` or `replace`.
+
+[id="plugins-{type}s-{plugin}-query_key"]
+===== `query_key`
+
+* Value type is <<string,string>>
+* Default value is `_id`.
+
+The key of the query to find the document to update or replace in MongoDB.
+
+query_key is used like described https://docs.mongodb.com/ruby-driver/v2.6/tutorials/ruby-driver-bulk-operations[here]
+for `update` and `replace` examples:
+
+  :filter => {query_key => query_value}
+
+[id="plugins-{type}s-{plugin}-query_value"]
+===== `query_value`
+
+* Value type is <<string,string>>
+* There is no default value for this setting.
+
+The value of the query to find the document to update or replace  in MongoDB. This can be dynamic using the `%{foo}` syntax.
+
+query_value is used like described https://docs.mongodb.com/ruby-driver/v2.6/tutorials/ruby-driver-bulk-operations[here]
+for `update` and `replace` examples:
+
+  :filter => {query_key => query_value}
```

Comment:
The usage of this is not really clear from this example, would you mind improving it a bit?

Reply:
Was a week on vacation. Will look into this asap.

Reply:
No problem take your time as you saw I went a bit beyond the original intent 😃

Hope you like the result

Reply:
It gets clear if you follow the link to mongodb documentation I added to ascii-documentation and take a look at update and replace examples.

Reply:
So in general you don't want to update a document in mongodb with whole logstash event but instead just increment some fields in mongodb? And do you want to read size to increment fields with from logstash event?

Reply:
Yeah I am building projections from the events and I am doing now something like:

```
    if [@metadata][event_type] == "stock.updated" and [quantity] > 0 {
        mongodb {
            id => "orders.projection.mongodb.stock-update"
            uri => "${MONGO_DNS}"
            collection => "product-aggregates"
            database => "carts"
            isodate => true
            action => "update"
            filter => {
                "_id" => "[_id]"
                "store_id" => "[store_id]"
            }
            update_expressions => {
                "$inc" => {"stock" => "[stock_delta]"}
            }
        }
    }
```